scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -764,11 +764,46 @@
                                        {:job-uuid->reserved-host (apply dissoc job-uuid->reserved-host matched-job-uuids)
                                         :launched-job-uuids (into matched-job-uuids launched-job-uuids)})))
 
+(defn distribute-task-requests-to-compute-clusters
```

Comment:
Buggy. This logic assumes that a given pool is in all compute clusters. 

What if a job's UUID happens to map to a compute cluster that cannot launch that pool. We'll auto-scale but be unable to ever launch.

I see a few choices here:
1. Detect if we're stuck and log with an error --- so if this occurs, we have a breadcrumb.
2. Scan all of the compute clusters and their nodes and remember for each pool what compute clusters have nodes in that pool, and distribute offers to only those pools. (this also depends that we have min-nodes-1 for any given pool on a compute cluster)

Reply:
Did we do any experiment to validate this distribution function? 

If the function has been validated, as part of future PRs:
- Maybe include smarter heuristics to avoid mapping jobs to compute-clusters we know will never schedule the jobs (e.g. k8s jobs to a mesos cluster). Maybe the compute cluster api needs to support a `pool-supported?` function.
- We should preferentially target pools (known from the job) which are low on resources as those are the ones which most likely need scaling up.

Out of curiosity, what scale's down the resources?

Reply:
@scrosby - Thanks! Yep, I agree that the distribution function needs to evolve, as described in the PR description (did you read it?). I would vote for this PR to keep it simple so that we can start to get value from autoscaling, and evolve it as those needs come up.

@shamsimam - You bring up a good point, which is that this currently will scale for Mesos pools in the KCCs, which is a bug. Thanks! I will address this.

Reply:
Still worth a TODO here. Is there an open ticket.

Reply:
Both issues mentioned above are now fixed -- the KCC's `:synthetic-pods` config now includes a `:pools` set, which specifies the set of pools for which the KCC should autoscale.